vspd.go

```diff
@@ -74,6 +82,31 @@ func run(ctx context.Context) error {
 	}
 	defer db.Close()
 
+	//Check if adminPass Hash is set in db.
+	hash, err := db.GetAdminHash()
+
+	//Ensure adminpass option is set
+	if cfg.AdminPass == "" && err != nil {
+		log.Error("Config error: the adminpass option is not set")
+		return err
+	}
+
+	if cfg.AdminPass != "" {
+		//Hash the cfg.AdminPass value
+		cfg.AdminPass, err = hashPassword(cfg.AdminPass)
+
+		if err != nil {
+			log.Errorf("Hashing AdminPass Failed: %v", err)
+			return err
+		}
+
+		//if adminpass is set, overwrite the saved adminpass hash in database.
+		db.UpdateAdminPass(cfg.AdminPass)
+
+	} else {
+		cfg.AdminPass = string(hash)
```

Comment:
I'll prefer to keep `cfg.AdminPass` clear/empty to ensure it is not set by the user somehow. Also, this variable doesn't seem to be accessed after here.